crates/provider/src/provider/eth_call/params.rs

```diff
@@ -52,18 +55,40 @@ where
     pub const fn block(&self) -> Option<BlockId> {
         self.block
     }
+
+    /// Sets the block overrides for this call.
+    pub fn with_block_overrides(mut self, overrides: BlockOverrides) -> Self {
+        self.block_overrides = Some(overrides);
+        self
+    }
+
+    /// Returns a reference to the block overrides if set.
+    pub const fn block_overrides(&self) -> Option<&BlockOverrides> {
+        self.block_overrides.as_ref()
+    }
 }
 
 impl<N: Network> serde::Serialize for EthCallParams<N> {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let len = if self.overrides().is_some() { 3 } else { 2 };
+        let len = match (self.block_overrides(), self.overrides()) {
+            (Some(_), Some(_)) => 4,
+            (Some(_), None) => 3,
+            (None, Some(_)) => 3,
+            (None, None) => 2,
+        };
```

Comment:
you can simplify this
we keep the previous one, but if blockoverrides are set then this is always  `len = 4`